bouncer/shared/executable_test.ts

```diff
@@ -0,0 +1,94 @@
+import { SwapContext } from './swap_context';
+import { runWithTimeout } from './utils';
+
+export enum TestStatus {
+  Ready = 'ready',
+  Running = 'running',
+  Complete = 'complete',
+  Failed = 'failed',
+}
+
+export class ExecutableTest {
+  public status: TestStatus = TestStatus.Ready;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private runFunction: (...args: any[]) => Promise<void>;
+
+  timeoutSeconds: number;
+
+  debug = false;
+
+  public swapContext: SwapContext;
+
+  constructor(
+    public name: string,
+    testFunction: () => Promise<void>,
+    timeoutSeconds: number,
+  ) {
+    this.runFunction = testFunction;
+    this.timeoutSeconds = timeoutSeconds;
+    this.swapContext = new SwapContext();
+  }
+
+  /// Run the test with the pre-defined timeout and any given arguments
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async run(...args: any[]) {
+    if (this.status === TestStatus.Running) {
+      throw new Error(`${this.name} Test is already running`);
+    }
+    console.log('\x1b[36m%s\x1b[0m', `=== Running ${this.name} test ===`);
```

Comment:
`\x1b[36m%s\x1b[0m` could we make this thing a constant of some kind () - it's not clear what this is by reading it

I understand the others are slightly different (it's not clear to me why they're different either), but we should be able to factor something out to make it more readable regardless.